common/persistence/nosql/nosql_execution_store_test.go

```diff
@@ -34,98 +34,114 @@ import (
 	"github.com/uber/cadence/service/history/constants"
 )
 
-func TestCreateWorkflowExecution(t *testing.T) {
+func TestNosqlExecutionStore(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 
 	mockDB := nosqlplugin.NewMockDB(ctrl)
 	logger := log.NewNoop()
-
-	store := &nosqlExecutionStore{
-		shardID:    1,
-		nosqlStore: nosqlStore{logger: logger, db: mockDB},
-	}
+	store := newTestNosqlExecutionStore(mockDB, logger)
 
 	ctx := context.Background()
-	request := &persistence.InternalCreateWorkflowExecutionRequest{
-		RangeID:                  123,
-		Mode:                     persistence.CreateWorkflowModeBrandNew,
-		PreviousRunID:            "previous-run-id",
-		PreviousLastWriteVersion: 456,
-		NewWorkflowSnapshot:      getNewWorkflowSnapshot(),
-	}
 	testCases := []struct {
 		name          string
 		setupMock     func()
-		request       *persistence.InternalCreateWorkflowExecutionRequest
-		expectedError error
+		testFunc      func() error // Use a function to encapsulate test execution
+		expectedError interface{}
 	}{
 		{
-			name: "success",
+			name: "CreateWorkflowExecution success",
```

Comment:
We should create `mockDB` and `store` per test case otherwise the proper isolation between test cases is not there. if one test fails for legitimate code change reasons and doesn't call the expected mock function then mock expectations will shift for subsequent tests. So let's make `mockDB` parameter to `testSetup` and create `store in the loop below not at the top.

Reply:
will do